### PR TITLE
Fix #699: Use one property for selecting userinfo bean

### LINKED ISSUE #699: Use one property for selecting userinfo bean
We use two boolean properties to pick the bean here:

https://github.com/wultra/enrollment-server/blob/develop/enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/UserInfoProviderConfiguration.java#L50

As a result, we can have a bean clash in case the following two properties are set to `true`:

```
enrollment-server.user-info.minimal-provider.enabled
enrollment-server.user-info.rest-provider.enabled
```

We should replace this property with an enum property, i.e.:

```
enrollment-server.user-info.provider=MINIMAL/REST
```

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

**Comment by zcgandcomp** [comment]
A question - originally we spoke about 3 levels

- No user-info
- Internal user-info
- proxied user-info

It would make more sense to me to have the following:

- one property (boolean) to turn the user-info - implicitly on the internal user-info level, default false,
- one property (string) to configure proxied call - explicitly switching to the proxied user-info.

This will be IMHO a more fool-proof configuration setup. 

The reviewed change will do a similar job. After the last config issues, I believe this will crash - at least in Azure without having a set-up default for `enrollment-server.user-info.provider`

File: docker/enrollment-server.xml [hunk 1]
```diff
@@ -40,8 +40,7 @@
     <Parameter name="enrollment-server.activation-spawn.enabled" value="${ENROLLMENT_SERVER_ACTIVATION_SPAWN_ENABLED}"/>
 
     <!--  User-info configuration  -->
-    <Parameter name="enrollment-server.user-info.minimal-provider.enabled" value="${ENROLLMENT_SERVER_USER_INFO_MINIMAL_ENABLED:false}"/>
-    <Parameter name="enrollment-server.user-info.rest-provider.enabled" value="${ENROLLMENT_SERVER_USER_INFO_REST_ENABLED:false}"/>
+    <Parameter name="enrollment-server.user-info.provider" value="${ENROLLMENT_SERVER_USER_INFO_PROVIDER:}"/>
```
Comment (zcgandcomp) [inline docker/enrollment-server.xml:43]: Last time the value without default crash the app during the startup. Please test it or put a default here.

**Review by zcgandcomp — CHANGES_REQUESTED** [review]

File: docker/enrollment-server.xml (hunk 1 above)
Comment (banterCZ) [reply docker/enrollment-server.xml:43]: I would like to see the error message. This is legit and working. It means no user-info provider at all. I have tested that. `${ENROLLMENT_SERVER_USER_INFO_PROVIDER:}` has a default value, which is empty.

**Review by romanstrobl — APPROVED** [review]
Looks OK.

**Review by zcgandcomp — APPROVED** [review]

**Pull request merged**